libraries/AP_L1_Control/AP_L1_Control.cpp

```diff
@@ -119,12 +119,60 @@ int32_t AP_L1_Control::target_bearing_cd(void) const
 }
 
 /*
-  this is the turn distance assuming a 90 degree turn
+  This is the turn distance assuming a 90 degree turn
  */
 float AP_L1_Control::turn_distance(float wp_radius) const
 {
-    wp_radius *= sq(_ahrs.get_EAS2TAS());
-    return MIN(wp_radius, _L1_dist);
```

Comment:
L1 distance is already scaled by the groundspeed, which will increase as you have more tailwind.
Isn't your approach double-dipping on the scaling of L1 distance?

I admit however that this `MIN(` is forcing a max distance of `WP_RADIUS`, which can be too small, and masking the effect of the L1 scaling.

What I would like to see is some testing that shows:
1. The current behaviour in head and tailwind.
2. Your proposed PR in head and tailwind.
3. What would happen if you turned this `MIN(` into a `MAX(`, instead of using your new code.

Reply:
Interesting Idea.
I can't say that I fully understand the L1 algo but with my limited knowledge I assume that L1 point will continue to update on the course and not stop at the waypoint.
The MIN(wp_radius, _L1_dist) seems to be an invariant for that.